packages/@storefront/flux-capacitor/src/core/sagas/refinements.ts

```diff
@@ -1,18 +1,41 @@
+import { Biasing, Sort } from 'groupby-api';
 import * as effects from 'redux-saga/effects';
 import FluxCapacitor from '../../flux-capacitor';
 import Actions from '../actions';
 import * as utils from '../actions/utils';
 import RecommendationsAdapter from '../adapters/recommendations';
 import RefinementsAdapter from '../adapters/refinements';
 import Events from '../events';
-import { refinementsRequest } from '../requests';
+import {
+  pastPurchaseProductsRequest,
+  refinementsRequest,
+} from '../requests';
 import Selectors from '../selectors';
 import Store from '../store';
 import RequestsTasks from './requests';
 
 export namespace RefinementsTasks {
   // tslint:disable-next-line max-line-length
-  export function* fetchMoreRefinements(flux: FluxCapacitor, { payload }: Actions.FetchMoreRefinements) {
+  export function* fetchMorePastPurchaseRefinements(flux: FluxCapacitor, { payload }: Actions.FetchMoreRefinements  | Actions.FetchMorePastPurchaseRefinements) {
+    try {
+      const pastPurchaseSkus: Store.PastPurchases.PastPurchaseProduct[] = yield effects.select(Selectors.pastPurchases);
+
+      if (pastPurchaseSkus.length > 0) {
+        const state: Store.State = yield effects.select();
+        const requestBody = pastPurchaseProductsRequest.composeRequest(state, payload.request);
+        const res = yield effects.call(RequestsTasks.refinements, flux, requestBody, payload.navigationId);
+
+        flux.emit(Events.BEACON_MORE_REFINEMENTS, payload.navigationId);
```

Comment:
I'm not sure if we should be beaconing this event here. @jmbtutor ?

Reply:
I think this beacon is associated with search

Reply:
Discussed via Slack; removing.